test_collect-logs.py

```diff
@@ -506,6 +556,19 @@ def test_get_units_failure(self):
         script.check_output.assert_not_called()
         script.call.assert_not_called()
 
+    def test_get_hosts_failure(self):
+        """
+        collect_logs() does not handle errors from get_hosts().
```

Comment:
Kept these as multi-line as it matched the majority of the rest of the file (although it's a bit mixed).